### ISSUE TITLE
Include broken links from content blocks

### ISSUE THREAD

**Issue by brynwhyman** [silverstripe-labs/silverstripe-externallinks#68]
## Overview
Currently this report will not include any external broken links that are included in elemental content blocks. As content blocks become a very popular method of creating content, this report should be tracking this content too.

We've seen a similar enhancement (with an open PR) to the elemental content blocks module to ensure that content is being included in the 'Broken page' and 'Broken links' reports, see https://github.com/dnadesign/silverstripe-elemental/issues/689

## ACs
* This looks through every block with a HTML field (including custom blocks)
* The report should continue to reference the _page_ that includes the content block with the broken link. There is no need to reference the block ID
* Running the report should result in broken links being highlighted in the WYSIWYG as 'broken'. Note this is existing behaviour for pages, so the aim is to match this functionality.
* Create a new issue to create a build task to retrospectively find broken links within existing blocks

## Notes
- This has been looked at before in https://github.com/silverstripe/silverstripe-externallinks/issues/37 (see issue for early investigation)

## PR's

- [ ] https://github.com/silverstripe/silverstripe-elemental/pull/822
- [ ] https://github.com/silverstripe/silverstripe-elemental/pull/861

**Comment by brynwhyman** [comment]
I still want to check if there's a UI change to broken links in a WYSIWYG after this report is run (i.e. the link is highlighted) as we may need an AC around this.

**Comment by robbieaverill** [comment]
Yes, it's supposed to highlight broken links with a red underline

**Comment by brynwhyman** [comment]
Thanks @robbieaverill. I've added another AC to reflect this:
> * Running the report should result in broken links being highlighted in the WYSIWYG as 'broken'. Note this is existing behaviour for pages, so the aim is to match this functionality.

**Comment by brynwhyman** [comment]
Related/ duplicate: https://github.com/silverstripe/silverstripe-externallinks/issues/37

**Comment by brynwhyman** [comment]
There's a similar issue which we've previously triaged as a bug: https://github.com/dnadesign/silverstripe-elemental/issues/689

**Comment by brynwhyman** [comment]
Noting that a recent bug was raised for CMS job run for this report. This should be reviewed when this issue is picked up: https://github.com/silverstripe/silverstripe-externallinks/issues/69
